.github/workflows/terraform.yml

```
@@ -11,25 +11,30 @@ jobs:
     name: "Determine Terraform Modules"
     runs-on: ubuntu-latest
     outputs:
-      modules: ${{ steps.output-var.outputs.modules }}
+      changed_modules: ${{ steps.output-var.outputs.changed_modules }}
+      all_modules: ${{ steps.output-var.outputs.all_modules }}
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
       - name: Pass to Var
         id: output-var
         env:
           ECHO_OUTPUT: ${{ steps.get-json.outputs.value }}
         run: |
-          JQ_OUTPUT=$(find modules/ -mindepth 2 -type d | jq -R -s -c 'split("\n")[:-1]')
-          echo "modules=$JQ_OUTPUT" >> $GITHUB_OUTPUT
+          JQ_OUTPUT_ALL=$(find modules/ -mindepth 2 -type d | jq -R -s -c 'split("\n")[:-1]')
+          JQ_OUTPUT_CHANGED=$(git diff --name-only ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} | grep -oP "modules/[^/]*/[^/]*" | sort -u | jq -R -s -c 'split("\n")[:-1]')
```

```suggestion
          JQ_OUTPUT_CHANGED=$(git diff --name-only ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }} | cut -d/ -f1-3 | sort -u | jq -R -s -c 'split("\n")[:-1]')
```

Weak suggestion avoiding a grep, though either would be fine

---

TIL